pkg/action/executor/move_util.go

```diff
@@ -0,0 +1,558 @@
+	_, err = rsClient.Update(rs)
+	if err != nil {
+		err = fmt.Errorf("failed to update RC-%v:%v\n", id, err.Error())
+		glog.Error(err.Error())
+		return currentName, err
+	}
+
+	return currentName, nil
+}
+
+//update the schedulerName of a ReplicationController
+// if condName is not empty, then only current schedulerName is same to condName, then will do the update.
+// return the previous schedulerName; or return "" if update failed.
+func updateRCscheduler(client *kclient.Clientset, nameSpace, rcName, schedulerName string) (string, error) {
+	currentName := ""
+
+	id := fmt.Sprintf("%v/%v", nameSpace, rcName)
+	rcClient := client.CoreV1().ReplicationControllers(nameSpace)
+
+	//1. get
+	option := metav1.GetOptions{}
+	rc, err := rcClient.Get(rcName, option)
+	if err != nil {
+		err = fmt.Errorf("failed to get ReplicationController-%v: %v\n", id, err.Error())
+		glog.Error(err.Error())
+		return currentName, err
+	}
+
+	if rc.Spec.Template.Spec.SchedulerName == schedulerName {
+		glog.V(3).Infof("no need to update schedulerName for RC-[%v]", rcName)
+		return "", nil
+	}
+
+	//2. update
+	rc.Spec.Template.Spec.SchedulerName = schedulerName
+	_, err = rcClient.Update(rc)
+	if err != nil {
+		err = fmt.Errorf("failed to update RC-%v:%v\n", id, err.Error())
+		glog.Error(err.Error())
+		return currentName, err
+	}
+
+	return currentName, nil
+}
+
+//---------------- for kubernetes server version <= 1.5 -------------------------------
+
+//provides some utilities for Kubernetes 1.5 and older
+const (
+	SchedulerAnnotationKey = "scheduler.alpha.kubernetes.io/name"
+	EmptyScheduler         = "None"
+)
+
+func getSchedulerName15(kclient *kclient.Clientset, kind, nameSpace, name string) (string, error) {
+	var annotes *map[string]string = nil
+
+	option := metav1.GetOptions{}
+	switch kind {
+	case kindReplicationController:
+		if rc, err := kclient.CoreV1().ReplicationControllers(nameSpace).Get(name, option); err == nil {
+			annotes = &(rc.Spec.Template.Annotations)
+		} else {
+			return "", err
+		}
+	case kindReplicaSet:
+		if rs, err := kclient.ExtensionsV1beta1().ReplicaSets(nameSpace).Get(name, option); err == nil {
+			annotes = &(rs.Spec.Template.Annotations)
+		} else {
+			return "", err
+		}
+	default:
+		return "", fmt.Errorf("unsported kind:[%v]", kind)
+	}
+
+	if *annotes == nil {
+		return EmptyScheduler, nil
+	}
+
+	result, ok := (*annotes)[SchedulerAnnotationKey]
+	if !ok || result == "" {
+		return EmptyScheduler, nil
+	}
+
+	return result, nil
+}
+
+func checkSchedulerName15(client *kclient.Clientset, kind, nameSpace, name, expectedScheduler string) (bool, error) {
+	currentName, err := getSchedulerName15(client, kind, nameSpace, name)
+	if err != nil {
+		return false, err
+	}
+
+	if currentName == expectedScheduler {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// shedulerName is set in annotations for kubernetes version < 1.6
+func updateAnnotedScheduler(pod *api.PodTemplateSpec, newName string) string {
```

Comment:
one minor suggestion, change the parameter name from `newName` to `newSchedulerName`.